tests/e2e/tests/skeleton/skeleton_test.go

```diff
@@ -0,0 +1,169 @@
+package skeleton_test
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	ec "github.com/cilium/tetragon/api/v1/tetragon/codegen/eventchecker"
+	sm "github.com/cilium/tetragon/pkg/matchers/stringmatcher"
+	"github.com/cilium/tetragon/tests/e2e/checker"
+	"github.com/cilium/tetragon/tests/e2e/helpers"
+	"github.com/cilium/tetragon/tests/e2e/runners"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+// This holds our test environment which we get from calling runners.NewRunner().Setup()
+var testenv env.Environment
+
+// The namespace where we want to spawn our pods
+const namespace = "skeleton"
+
+// Every e2e test needs a TestMain with the two lines below. The runner supports
+// customization options that can be used to override its setup behaviour, but in most
+// cases the defaults will be enough combined with command line flags.
+//
+// See tests/e2e/flags/flags.go for a detailed overview of what Tetragon-specific command
+// line flags can be used.
+func TestMain(m *testing.M) {
+	// Set up the tests. In a default configuration, this does the following:
+	//
+	// 1. At the beginning of every test, check to see whether we have a kubeconfig file
+	//    (specified at the command line with -kubeconfig). If not, create a temporary
+	//    kind cluster to run the tests.
+	//
+	// 2. After the cluster is configured and running, query the minimum kernel versions
+	//    supported by all nodes and set this as a variable in the test context.
+	//
+	// 3. Register a hook at the start of every test that installs Cilium into the
+	//    cluster with some default options (unless -tetragon.install-cilium=false is set
+	//    on thhe command line).
+	//
+	// 4. Register a hook at the start of every test that installs Tetragon into the
+	//    cluster with some default options.
+	//
+	// 5. Register a hook at the start of every test that port forwards Tetragon metrics
+	//    and gRPC ports for all pods. These port forwards are registered in the test
+	//    context for later retrieval.
+	//
+	// 6. Register a hook at the end of every test that dumps information about the
+	//    cluster and running event checkers. This information is only dumped if the test
+	//    fails or if -tetragon.keep-export=true is set on the command line.
+	//
+	testenv = runners.NewRunner().Setup()
+
+	// Any additional setup and cleanup can be performed here if you like.
+	// This would be done using testenv.Setup() and testenv.Finish() respectively.
+	//
+	// These take a list of functions to register at test start and test end that can be
+	// used to maniplate the test context and test environment.
+	//
+	// Here we ensure our test namespace doesn't already exist then create it.
+	testenv.Setup(func(ctx context.Context, c *envconf.Config) (context.Context, error) {
+		ctx, _ = helpers.DeleteNamespace(namespace, true)(ctx, c)
+
```

Comment:
Just to get this right: overall we can have parallel tests and they won't clash as long as they run in different namespace? and it is developer responsibility to make sure that he is reusing the same namespace between DeleteNamespace() and tests running ?
Thinking if we should remove DeleteNamespace() here and if the test env exists and it fails we know maybe something is wrong on how they are setup... better than having tests stopped or something disappearing and trying to figure it out... does this makes sense ?

Reply:
> Just to get this right: overall we can have parallel tests...

In theory yes, but unfortunately due to the way the Cilium installation helpers are written on the Cilium side you can't run tests in parallel at the moment. Once we resolve those issues up stream then yes, it should be possible.

> ... and they won't clash as long as they run in different namespace?

Right. Also thinking about running the same test back to back here. For example you could imagine running with -count 3 or something.

> ...and it is developer responsibility to make sure that he is reusing the same namespace between DeleteNamespace() and tests running?

Another option is to just use whatever namespace e2e-framework gives you. There is definitely a trade-off here in terms of idempotency vs too much magic. I lean toward giving the test author control over what namespace(s) to use in the test though.

> Thinking if we should remove DeleteNamespace() here and if the test env exists and it fails we know maybe something is wrong on how they are setup

That's how I had it initially but it breaks the case where you want to run tests back to back e.g. with -count (see above).

Let me know if the above makes sense to you. Happy to have my mind changed on any of these points if folks feel strongly one way or the other.

Reply:
Sounds good thank you @willfindlay